### DISCUSSION TITLE
cli: add export to SVG animation

### DISCUSSION THREAD

**Discussion by brunoborges** [asciinema/discussions#127] [Ideas]
The project termtosvg is interesting, but I believe adding that functionality as an option in the CLI of 
asciinema would make more sense.

httpsithub.com/nbedos/termtosvg

**Comment by ku1ik** [comment]
There's also https://github.com/marionebl/svg-term-cli by @marionebl.

I'm not sure if this functionality should be in the recorder itself.

I totally understand it would be convenient, at least for a subset of users though. I see `asciinema` (cli) as a tool which can record and replay terminal sessions, within terminal. It basically deals with asciicast files, and terminal. 

It's also a non-trivial amount of code to maintain, and I prefer to not do that :)

Having said that, I could see a way to integrate the mentioned tools without keeping this functionality in the core: "converters".

Example:

You would put this in asciinema config file:

```ini
[converters]
svg = "termtosvg render %1 %2"
gif = "asciicast2gif %1 %2"
```

Then record to SVG with:

```bash
asciinema rec demo.svg
```

The session would be recorded to tmp asciicast file, then converted automatically by calling one of the defined converter commands (the one which matches the desired file's extension).

You would still need to install the extra tool, but other than that it could be quite smooth experience.

**Comment by brunoborges** [comment]
Hi @sickill, I like your suggested approach very much and as a user I'd be OK with it. It brings the convenience without the burden of maintenance. 

👍

**Comment by Gowee** [comment]
I have a naive proposal:

Just write a wrapper around the svg-term project so that to deploy it onto a serverless platform such as [Vercel](https://vercel.com/dashboard), exposing a simple web API that converts an asciicast file into a SVG. Then add an "Export as SVG" button on https://asciinema.org/ to redirect users to that API endpoint (or serve the SVG directly by internally calling the API server-side).

In this way, there is no need for users to install svg-term and Node.js along with tons of modules locally, which is fairly daunting. Also the existing asciinema CLI requires no change and hence no much extra maintainence effort.

**Comment by ciq-mcarifio** [comment]
Is the `[converter]` proposal actually implemented? I'm guessing no (since I just tried it).
